### ISSUE TITLE
error : running wasmer aot file

### ISSUE THREAD

**Issue by Franky45** [wasmerio/wasmer#5006]
![image (6)](https://github.com/user-attachments/assets/cac1a363-1c7d-44ab-b92e-ec5c000956c1)

 
### Describe the bug
 
```sh
wasmer -vV: 
wasmer 4.3.1 (d5df452 2024-05-23)
binary: wasmer-cli
commit-hash: d5df452d22110593f12fcc0e152a0435344b6590
commit-date: 2024-05-23
host: x86_64-unknown-linux-gnu
compiler: singlepass,cranelift,llvm

rustc -vV:
rustc 1.80.1 (3f5fd8dd4 2024-08-06)

binary: rustc

commit-hash: 3f5fd8dd41153bc5fdca9427e9e05be2c767ba23

commit-date: 2024-08-06

host: x86_64-unknown-linux-gnu

release: 1.80.1

LLVM version: 18.1.7

```
 
 
### Steps to reproduce
<!--
Include steps that will help us recreate the issue.
 
to compile wasmer to aot 
$ wasmer compile hi.wasm -o hi.wasmc
 
To run :
 
 
 
If applicable, add a link to a test case (as a zip file or link to a repository we can clone).
-->
 
### Expected behavior
<!-- A clear and concise description of what you expected to happen. -->
 
### Actual behavior
 
<!--
A clear and concise description of what actually happened.
 
If applicable, add screenshots to help explain your problem.
-->
 
### Additional context
<!-- Add any other context about the problem here. -->

in this format upload ypur bug
 
 
for the bug report! -->
 
### Describe the bug
 
<!--
After compiling wamser to aot (i.e, wasmc), when running the .wasmc  i am getting error 
 
 attach sccrenshot
 
Copy and paste the result of executing the following in your shell, so we can know the version of wasmer, Rust (if available) and architecture of your environment.
-->```sh
wasmer -vV: 
wasmer 4.3.1 (d5df452 2024-05-23)
binary: wasmer-cli
commit-hash: d5df452d22110593f12fcc0e152a0435344b6590
commit-date: 2024-05-23
host: x86_64-unknown-linux-gnu
compiler: singlepass,cranelift,llvm

rustc -vV:
rustc 1.80.1 (3f5fd8dd4 2024-08-06)

binary: rustc

commit-hash: 3f5fd8dd41153bc5fdca9427e9e05be2c767ba23

commit-date: 2024-08-06

host: x86_64-unknown-linux-gnu

release: 1.80.1

LLVM version: 18.1.7

```
 
 
### Expected behavior

 
Should execute the .wasmc file 
 
### Actual behavior
 

error: module hash is not present in the artifact

**Comment by zhangchaoza** [comment]
I had the same problem. I checked the code and I guess that the hash option was not set to engine in fact, it was just set to the clone of the engine. So, the module is missing its hash.

https://github.com/wasmerio/wasmer/blob/56894ff7b73dde5555675a8b11ce75363281b2a7/lib/cli/src/commands/compile.rs#L62-L64

**Comment by maminrayej** [comment]
Closing as the fix for this should be in the 4.3.6 release.
